supernode/services/senseregister/task.go

```diff
@@ -559,7 +546,27 @@ func (task *Task) ValidateAndRegister(ctx context.Context, ticket []byte, creato
 	return nftRegTxid, err
 }
 
-func (task *Task) waitConfirmation(ctx context.Context, txid string, minConfirmation int64, interval time.Duration) <-chan error {
+func (task *Task) ValidateActionActAndStore(ctx context.Context, actionRegTxId string) error {
+	var err error
+
+	// Wait for action ticket to be activated by walletnode
+	confirmations := task.waitActionActivation(ctx, actionRegTxId, 2, 30*time.Second)
+	err = <-confirmations
+	if err != nil {
+		return errors.Errorf("wait for confirmation of reg-art ticket %w", err)
+	}
+
+	// Store dd_and_fingerprints into Kademlia
+	if err = task.storeIDFiles(ctx); err != nil {
+		log.WithContext(ctx).WithError(err).Errorf("store id files")
+		err = errors.Errorf("store id files: %w", err)
+		return nil
+	}
+
```

Comment:
above functions will take long time to finish, it better to run it in `<-task.NewAction` context